addons/web/static/src/js/views/graph/graph_renderer.js

```diff
@@ -1,1079 +1,1049 @@
+    const _t = core._t;
+    const qweb = core.qweb;
+
+    const NO_DATA = [_t('No data')];
+    NO_DATA.isNoData = true;
+    const CHART_TYPES = ['pie', 'bar', 'line'];
+
+    const COLORS = ["#1f77b4", "#ff7f0e", "#aec7e8", "#ffbb78", "#2ca02c", "#98df8a", "#d62728",
+        "#ff9896", "#9467bd", "#c5b0d5", "#8c564b", "#c49c94", "#e377c2", "#f7b6d2",
+        "#7f7f7f", "#c7c7c7", "#bcbd22", "#dbdb8d", "#17becf", "#9edae5"];
+    const COLOR_NB = COLORS.length;
+
+    // used to format values in tooltips and yAxes.
+    const FORMAT_OPTIONS = {
+        // allow to decide if utils.human_number should be used
+        humanReadable: value => {
+            return Math.abs(value) >= 1000;
+        },
+        // with the choices below, 1236 is represented by 1.24k
+        minDigits: 1,
+        decimals: 2,
+        // avoid comma separators for thousands in numbers when human_number is used
+        formatterCallback: str => {
+            return str;
+        },
+    };
+
+    // hide top legend when too many items for device size
+    const MAX_LEGEND_LENGTH = 4 * (Math.max(1, config.device.size_class));
+
+    const {useState, useRef } = owl.hooks;
+    class GraphOwlRenderer extends OwlAbstractRenderer {
+
+        constructor() {
+            super(...arguments);
+            this.noContentHelperData = {show: false, title: "", description: ""};
+            this.canvasRef = useRef("canvas");
+
+            this.filteredDataPoints = this._filterDataPoints(this.props);
+            this._setNoContentHelper(this.props);
         }
-    },
-    /**
-     * This function creates a custom HTML tooltip.
-     *
-     * @private
-     * @param {Object} tooltipModel see chartjs documentation
-     */
-    _customTooltip: function (tooltipModel) {
-        this.$el.css({ cursor: 'default' });
-        if (this.$tooltip) {
-            this.$tooltip.remove();
+
+        mounted() {
+            this._renderChart();
         }
-        if (tooltipModel.opacity === 0) {
-            return;
+
+        async willUpdateProps(nextProps) {
+            this.filteredDataPoints = this._filterDataPoints(nextProps);
+            this._setNoContentHelper(nextProps);
         }
-        if (tooltipModel.dataPoints.length === 0) {
-            return;
+
+        patched() {
+            this._renderChart();
         }
 
-        if (this._isRedirectionEnabled()) {
-            this.$el.css({ cursor: 'pointer' });
+        //--------------------------------------------------------------------------
+        // Private
+        //--------------------------------------------------------------------------
+
+        /**
+         * This function aims to remove a suitable number of lines from the tooltip in order to make it reasonably visible.
+         * A message indicating the number of lines is added if necessary.
+         *
+         * @private
+         * @param {Number} maxTooltipHeight this the max height in pixels of the tooltip
```

Comment:
?

Reply:
Do you mean the "A message indicating the number of lines is added if necessary"?

I wrote "An ellipsis is added in case lines are actually removed" instead.